packages/contracts-bedrock/src/cannon/PreimageOracle.sol

```diff
@@ -183,6 +183,81 @@ contract PreimageOracle is IPreimageOracle {
         preimageLengths[key] = size;
     }
 
+    /// @inheritdoc IPreimageOracle
+    function loadBlobPreimagePart(
+        uint256 _z,
+        uint256 _y,
+        bytes calldata _commitment,
+        bytes calldata _proof,
+        uint256 _partOffset
+    )
+        external
+    {
+        uint256 size;
+        bytes32 key;
+        bytes32 part;
+        assembly {
+            // Compute the versioned hash. The SHA2 hash of the 48 byte commitment is masked with the version byte,
+            // which is currently 1. https://eips.ethereum.org/EIPS/eip-4844#parameters
+            calldatacopy(0x00, _commitment.offset, 0x30)
+            let success := staticcall(gas(), 0x02, 0x00, 0x30, 0x00, 0x20)
+            if iszero(success) {
+                // Store the "ShaFailed()" error selector.
+                mstore(0x00, 0xf9112969)
+                // revert with "ShaFailed()"
+                revert(0x1C, 0x04)
+            }
+            let versionedHash := or(and(mload(0x00), not(shl(248, 0xFF))), shl(248, 1))
+
+            // we leave solidity slots 0x40 and 0x60 untouched, and everything after as scratch-memory.
+            let ptr := 0x80
+
+            // Load the inputs for the point evaluation precompile into memory.
+            mstore(ptr, versionedHash)
+            mstore(add(ptr, 0x20), _z)
+            mstore(add(ptr, 0x40), _y)
+            calldatacopy(add(ptr, 0x60), _commitment.offset, 0x30)
+            calldatacopy(add(ptr, 0x90), _proof.offset, 0x30)
+
+            // Verify the KZG proof.
+            success := staticcall(gas(), 0x0A, ptr, 0xC0, 0x00, 0x00)
+            if iszero(success) {
+                // Store the "InvalidProof()" error selector.
+                mstore(0x00, 0x09bde339)
+                // revert with "InvalidProof()"
+                revert(0x1C, 0x04)
+            }
+
+            // len(sig) + len(partOffset) + len(preimage offset) = 4 + 32 + 32 = 0x44
+            size := calldataload(0x44)
```

Comment:
What's happening here? Isn't the `size` of a field-element a constant `0x20` in bytes? (excl. any length prefix data).

Reply:
ah yikes - had fixed this and didn't push, nice spot.